onnx_tf/handlers/backend/tfidf_vectorizer.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+import tensorflow as tf
+
+from onnx_tf.handlers.backend_handler import BackendHandler
+from onnx_tf.handlers.handler import onnx_op
+from onnx_tf.handlers.handler import tf_func
+
+
+@onnx_op("TfIdfVectorizer")
+class TfIdfVectorizer(BackendHandler):
+
+  @classmethod
+  def args_check(cls, node, **kwargs):
+    if "pool_int64s" in node.attrs and "pool_strings" in node.attrs:
+      raise ValueError(
+          "Cannot set the pool_int64s and pool_strings in an input at the same time."
+      )
+
+  @classmethod
+  def _prepare_ngrams(cls, x, n, skip):
+    # This method transform input into n-grams for a specific skip
+    # input x is a 1D tensor
+    # ex1: x=[1,2,3,4] n=1 skip=0 output=[[1],[2],[3],[4]]
+    # ex2: x=[1,2,3,4] n=2 skip=0 output=[[1,2],[2,3],[3,4]]
+    # ex3: x=[1,2,3,4] n=2 skip=1 output=[[1,3],[2,4]]
+    count = x.shape[0] - n + 1 - skip
+    mul = skip + 1
+    ngrams = [x[i * mul:i * mul + count] for i in range(n)]
+    ngrams = tf.stack(ngrams)
+    ngrams = tf.transpose(ngrams, [1, 0])
+    return ngrams
+
+  @classmethod
+  def _calc_ngram(cls, x, pool, n, max_skip):
+    # This method calculates ngram counts for a specific n and
+    # all allowable skips
+
+    # For 1gram, skip is not in use
+    if n == 1:
+      return cls._calc_ngram_skip(x, pool, n)
```

Comment:
I think it may be more appropriate to have a special case in `_prepare_ngrams` for better abstraction. Because I think you actually need this special case for this whole algorithm to work correctly, which implies that `_prepare_ngrams`'s abstraction breaks when n=1 and skip is not zero, meaning `_prepare_ngrams` is not really self-contained.

Reply:
Please check test_node.py, # test IDF mode with weights, max_skip=5, output contains 1-grams and 2-grams. It should invoke _prepare_ngrams with n=1 and skip=5. Maybe I missed something?

Reply:
I read the spec one more time and don't see "For 1gram, skip is not in use". However the ONNX examples seem to indicate so. As seen in tf_batch_uniandbigrams_skip5 and tf_uniandbigrams_skip5, the 1-gram results [0, 3, 0, 0] and [0, 3, 1, 0] are not the accumulated counts from multiple skips. Therefore the special handling for 1gram is based on examples.